### PR TITLE
Add time integration scheme to the default console header output

### LINKED ISSUE #23653: Add time integration scheme to the output
## Reason
Different codes have different defaults now. This could help distinguish what's going on between different logs
It's also a very important simulation parameter overall, just as important as the spatial scheme

## Design
Add a lign in the header log

## Impact
More output, which is not great
But more descriptive logs, which can be great for analyzing a simulation

### PR DESCRIPTION
**Pull request by GiudGiud**
closes #23653

### CODE REVIEW

File: framework/src/executioners/Transient.C [hunk 1]
```diff
@@ -741,6 +708,33 @@ Transient::getTimeStepperName()
     return std::string();
 }
 
+std::string
+Transient::getTimeIntegratorName()
+{
```
Comment (YaqiWang) [inline framework/src/executioners/Transient.C:713]: I like to have this on screen output. Should we make this a function in `TimeIntegrator::type()` ? In that way, we can just return `_ti->type()` here. We actually had a function `Transient::getTimeScheme()` returns `Moose::TimeIntegratorType`. If we have a stringfy function, we can use that function here as well. The reason I am asking is that there could be more time integrators.

Comment (GiudGiud) [reply framework/src/executioners/Transient.C:713]: can do that

Comment (GiudGiud) [reply framework/src/executioners/Transient.C:713]: I m not sure this well help actually but made the change. it s consistent with other enums anyway

Comment (GiudGiud) [reply framework/src/executioners/Transient.C:713]: nevermind it works with type().

stringify would not work for your new types (outside of MOOSE) of time integrators

File: framework/src/executioners/Transient.C [hunk 2]
```diff
@@ -741,6 +741,16 @@ Transient::getTimeStepperName()
...
 }
 
+std::string
+Transient::getTimeIntegratorName() const
+{
+  const auto * ti = _nl.getTimeIntegrator();
+  if (ti)
+    return ti->type();
+  else
+    mooseError("Time integrator has not been built yet so we can't retrieve its name");
```
Comment (loganharbour) [inline framework/src/executioners/Transient.C:751]:
```suggestion
  if (const auto * ti = _nl.getTimeIntegrator())
    return ti->type();
  return {};
```

I don't think it's necessary to error here - your call

**Review by loganharbour — APPROVED** [review]

File: framework/src/executioners/Transient.C (hunk 2 above)
Comment (GiudGiud) [reply framework/src/executioners/Transient.C:751]: yeah either/or. with an error we re sure to never print the wrong thing

**Comment by doquang** [comment]
@GiudGiud - The new line you added for this ticket is incorrect. It should be
  `if (time_integrator != "")`, not   `if (time_stepper != "")`
```
std::string
outputExecutionInformation(const MooseApp & app, FEProblemBase & problem)
{

  std::stringstream oss;
  oss << std::left;

  Executioner * exec = app.getExecutioner();

  oss << "Execution Information:\n"
      << std::setw(console_field_width) << "  Executioner: " << demangle(typeid(*exec).name())
      << '\n';

  std::string time_stepper = exec->getTimeStepperName();
  if (time_stepper != "")
    oss << std::setw(console_field_width) << "  TimeStepper: " << time_stepper << '\n';
  std::string time_integrator = exec->getTimeIntegratorName();
  if (time_stepper != "")
    oss << std::setw(console_field_width) << "  TimeIntegrator: " << time_integrator << '\n';

  oss << std::setw(console_field_width) << "  Solver Mode: " << problem.solverTypeString() << '\n';

  const std::string & pc_desc = problem.getPetscOptions().pc_description;
  if (!pc_desc.empty())
    oss << std::setw(console_field_width) << "  PETSc Preconditioner: " << pc_desc << '\n';

  MoosePreconditioner const * mpc = problem.getNonlinearSystemBase().getPreconditioner();
  if (mpc)
  {
    oss << std::setw(console_field_width)
        << "  MOOSE Preconditioner: " << mpc->getParam<std::string>("_type");
    if (mpc->name() == "_moose_auto")
      oss << " (auto)";
    oss << '\n';
  }
  oss << std::endl;

  return oss.str();
}
```

**Comment by GiudGiud** [comment]
Good catch I ll fix it

**Pull request merged**